### PR TITLE
KOGITO-2178 Implement customize column visibility functionality for the Generic T…

### PR DESCRIPTION
**Pull request by yzhao583**
Jira: https://issues.redhat.com/browse/KOGITO-2178

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by pefernan — CHANGES_REQUESTED** [review]
@yzhao583 code looks well but it seems that there's a conflict, would you mind rebasign?

thanks!

**Comment by yzhao583** [comment]
@pefernan Updated Thanks!

**Comment by cristianonicolai** [comment]
@yzhao583 @pefernan @nmirasch  wouldn't this be something that could be reused across consoles?

**Comment by pefernan** [comment]
@cristianonicolai yea, but since this won't be finished by devpreview 2 I asked @yzhao583 to add it on Task Console to keep the common module clean. My fault here.

**Review by pefernan — APPROVED** [review]
Waiting for @jstastny-cz but I'd say +1

File: packages/task-console/src/components/Templates/DataTableContainer/tests/__snapshots__/DataTableContainer.test.tsx.snap [hunk 1]
```diff
@@ -171,15 +171,61 @@ exports[`DataTableContainer component tests Snapshot tests 1`] = `
...
+                          ],
+                          "data": "state",
+                          "title": "State",
+                        },
                       ]
                     }
-                    isLoading={true}
-                    networkStatus={1}
+                    error={
+                      [Error: Network error: No more mocked responses for the query: query getUserTasksByStates($state: [String!]) {
```
Comment (jstastny-cz) [inline packages/task-console/src/components/Templates/DataTableContainer/tests/__snapshots__/DataTableContainer.test.tsx.snap]: here it seems like an error in test setup, or is the failure expected?

Comment (jstastny-cz) [reply packages/task-console/src/components/Templates/DataTableContainer/tests/__snapshots__/DataTableContainer.test.tsx.snap]: Probably due to not passing mocks to the MockedProvider.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

File: packages/task-console/src/components/Templates/DataTableContainer/tests/__snapshots__/DataTableContainer.test.tsx.snap (hunk 1 above)
Comment (yzhao583) [reply packages/task-console/src/components/Templates/DataTableContainer/tests/__snapshots__/DataTableContainer.test.tsx.snap]: @jstastny-cz Updated. Thanks!

**Review by jstastny-cz — APPROVED** [review]

**Comment by yzhao583** [comment]
@cristianonicolai @pefernan @jstastny-cz Hi guys, is there anything else I need to fix? If not, can we merge this PR so we can move forward? Thanks!

**Pull request merged**